File: stop_gninnips_my_sdrow/stop_gninnips_my_sdrow.java
import java.util.stream.Collectors;
import java.util.Arrays;
import java.lang.StringBuilder;

public class SpinWords {
    public String spinWords(String sentence) {
        return Arrays.asList(sentence.split(" "))
            .stream()
            .map(v -> v.length() >= 5 ? new StringBuilder(v).reverse().toString() : v)
            .collect(Collectors.joining(" "));
    }
}

File: stop_gninnips_my_sdrow/stop_gninnips_my_sdrow.rs
fn spin_words(words: &str) -> String {
    words
        .split_whitespace()
        .map(|x| if x.len() >= 5 { x.chars().rev().collect::<String>() } else { x.to_string() })
        .collect::<Vec<String>>()
        .join(" ")
}
